src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1431,6 +1431,12 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return connectionProvider && connectionProvider.properties;
 	}
 
+	// Gets languageMode property of provider if it exists. Defaults to 'sql'
+	public getProviderLanaguageMode(providerName: string): string {
+		let connectionProvider = this._providers.get(providerName);
+		return (connectionProvider && connectionProvider.properties && connectionProvider.properties['languageMode']) || 'sql';
```

Comment:
`return this._providers.get(providerName)?.properties?.['languageMode'] || 'sql';`